### DISCUSSION TITLE
Enable selective packet capture

### DISCUSSION THREAD

**Discussion by logz4u** [Security-Onion-Solutions/securityonion#10137] [Ideas]
Good morning!

Analyzing TLS handshakes is becoming more of a common thing, especially in the machine learning sector. However, capturing full PCAP is EXPENSIVE, and I am very grateful for steno tuning. Would it be possible to enable an additional filter where stenographer could capture the first...20 or 30 or whatever packets of a session? Just enough to determine the parameters of the TLS handshake.  

If this is already possible, my apologies!

**Comment by TOoSmOotH** [comment]
Steno is not protocol aware. It just writes and indexes packets. 

see #9131

**Reply by logz4u** [reply]
Oh gee, I knew that, oops. How would Steno know if it was seeing a handshake in the first place! Sorry about that. 

"Resolved" from my POV.
